pkg/collectors/pod.go

```diff
@@ -382,21 +431,30 @@ func (pc *podCollector) collectPod(ch chan<- prometheus.Metric, p v1.Pod) {
 
 	var lastFinishTime float64
 
-	for _, cs := range p.Status.ContainerStatuses {
-		addGauge(descPodContainerInfo, 1,
+	collectContainerStatusMetric := func(cs v1.ContainerStatus, info *prometheus.Desc, waiting *prometheus.Desc, _waitingReason *prometheus.Desc, running *prometheus.Desc, terminated *prometheus.Desc, terminatedReason *prometheus.Desc, restarts *prometheus.Desc) {
```

Comment:
How about instead of passing in each metric description as arguments to the function, just passing in a single boolean e.g. `isInitContainer`. The function then decides which metric descriptions to use based on that. Should make this less verbose.

Reply:
I agree it would reduce the params but not sure if that make it less verbose. 
``` 
if init {
desc1 = init_metrics_desc1
...
}
else {
desc1 = metrics_desc1
...
}
addGauge(metrics_desc1, 1)
```
If i can make the change if you strongly feel about it

Reply:
What do others think?

Reply:
how about a third way? we pass a map instead of each single description, that way we have a single parameter and we can initialize the map just like the descriptions

Reply:
Sounds good. I made the change.